octane/evmengine/keeper/keeper.go

```diff
@@ -270,3 +276,20 @@ func (k *Keeper) getOptimisticPayload() (engine.PayloadID, uint64, time.Time) {
 
 	return k.mutablePayload.ID, k.mutablePayload.Height, k.mutablePayload.UpdatedAt
 }
+
+func withdrawalsEqual(w1, w2 []*etypes.Withdrawal) bool {
+	if len(w1) != len(w2) {
+		return false
+	}
+
+	for i, w := range w1 {
+		if w.Index != w2[i].Index ||
```

Comment:
the fields of the withdrawal are all comparable, so I think you can do:
```
if w1 == nil || w2 == nil {
  return false
} else if *w1 != *w2 {
  return false
}
```

Also remember to write a unit test for this function to ensure malicious proposer cant slip in invalid withdrawals.